src/traits.rs

```diff
@@ -59,3 +59,18 @@ impl<'a,T> Strided for MutStride<'a,T> {
 impl<'a,T> MutStrided for MutStride<'a,T> {
     fn as_stride_mut(&mut self) -> MutStride<T> { self.reborrow() }
 }
+
+// this isn't as general as it could be.
+impl<'a,T, X: Deref<Target=[T]>> From<&'a X> for Stride<'a,T> {
```

Comment:
Is `AsRef` automatic for `Deref`? Or will there be cases where `AsRef` is not implemented but `Deref` is?